x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -266,6 +266,17 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 				details["error"] = err.Error()
 				return errorLogAndFormat("relay_payment_failed_project_add_cu", details, "Failed to add CU to the project")
 			}
+
+			project, _, err := k.projectsKeeper.GetProjectForDeveloper(ctx, clientAddr.String(), uint64(ctx.BlockHeight()))
+			if err != nil {
+				details["error"] = err.Error()
+				return errorLogAndFormat("relay_payment_failed_getting_project", details, "Failed to get project for developer")
+			}
+			err = k.subscriptionKeeper.AddComputeUnitsToSubscription(ctx, project.GetSubscription(), relay.CuSum)
```

Comment:
Check that the project/subscription still have enough CUs left?

Reply:
I don't think that this is the right place for this check. The provider should check if there's enough CU **before** he provides service.